### DISCUSSION TITLE
Force draw without showing view

### DISCUSSION THREAD

**Discussion by arunningpir8** [Live-Charts/LiveCharts2#558] [General]
Hello!

I have a use case where I used to use the v0 live chart but it had a huge memory leak (that doesn't happen in v2).  However, I need to be able to save an image of the chart amongst other information from a view in memory using a ViewBox.  This was possible in the old v0 by calling:
` Chart.Update(true, true); //force chart redraw`

however using the new update method:
`Chart.CoreChart.Update(new LiveChartsCore.Kernel.ChartUpdateParams() { IsAutomaticUpdate = false, Throttling = false });`

does not work the same, and causes a blank canvas when saving, and the UpdateFinished is never fired.  Using the SKCartesianChart.SaveImage() function isn't really useful in this case either as it doesn't show legend or any of the other information in the view I want.

Is there currently a way to force a draw of the chart without calling Window.Show() and with using a ViewBox instead?

**Comment by skerr0** [comment]
I am using a hidden window to render the charts in the background, but obviously this solution is extremely poor. 

It looks like at some point there may have been a way to save a chart image? Is there a reason this was scrapped? Would it be reasonable to add a way to skip animation and immediately draw the final chart for export if not?

**Reply by arunningpir8** [reply]
I am also currently using a hidden window by setting `window.Top = -50000;`, but it is obviously not ideal and just kind of "messy".  It's just frustrating that this is literally the only thing preventing this from being a perfect replacement in code from v0, especially because v2 has a save function, but I need it for my client to be amongst other information.  I even tried stitching pictures together, but it always comes out "noticeable" which is worse than just doing the window hack, plus the save image function doesn't include legend.
